framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -311,8 +422,168 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 		return false;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	private async _createCCU() {}
+	private async _calculateInboxUpdate(sideChainID: Buffer): Promise<InboxUpdate> {
+		const serialzedCCMs = this._chainConnectorState.crossChainMessages.map(ccm =>
+			codec.encode(ccmSchema, ccm),
+		);
+		const ccmHashes = serialzedCCMs.map(ccm => utils.hash(ccm));
+
+		const channel = await this._mainchainAPIClient.invoke<ChannelData>(
+			'interoperability_getChannel',
+			{ sideChainID },
+		);
+
+		let newInboxRoot;
+		let newInboxAppendPath = channel.inbox.appendPath;
+		let newInboxSize = channel.inbox.size;
+
+		for (const ccm of ccmHashes) {
+			const { appendPath, size, root } = regularMerkleTree.calculateMerkleRoot({
+				value: ccm,
+				appendPath: newInboxAppendPath,
+				size: newInboxSize,
+			});
+			newInboxAppendPath = appendPath;
+			newInboxSize = size;
+			newInboxRoot = root;
+		}
+
+		const db = new InMemoryDatabase();
+
+		const smt = new SparseMerkleTree({
```

Comment:
I am very confused by this line. Why are we initializing the SMT with the inbox root as root hash?

Reply:
Actually I'm just confused about what are you trying to do here. You create a new SMT from an empty DB, passing a non-empty root? Then you update the tree, which should overwrite the root anyway. Then how can you use that SMT to generate a proof? isn't the only key you inserted just the root?

Reply:
oh instead of in-memory db it should've been stateDB, need to find a way to access the stateDB through plugin or we can create an endpoint to generate proof on application